internal/controllers/clusterextension_controller.go

```diff
@@ -286,18 +291,17 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 	managedObjs, _, err := r.Applier.Apply(ctx, unpackResult.Bundle, ext, objLbls, storeLbls)
 	if err != nil {
 		setStatusProgressing(ext, wrapErrorWithResolutionInfo(resolvedBundleMetadata, err))
-		// If bundle is not already installed, set Installed status condition to False
-		if installedBundle == nil {
-			setInstalledStatusConditionFailed(ext, err.Error())
-		}
+		// Now that we're actually trying to install, use the error
+		setInstalledStatusFromBundle(ext, installedBundle, err)
```

Comment:
Yesterday we said:
- Failed install
   - status.conditions[Installed] = false
- Failed upgrade
   - status.conditions[Installed] = true, message indicates successfully installed bundle ref

So should this be:

```suggestion
		// Now that we're actually trying to install, use the error
		setInstalledStatusFromBundle(ext, installedBundle, nil)
```

So that the `Installed`

Reply:
No, we want the `err` in there, the code is correct.

A failed (initial) install means that `installedBundle` is nil, and `err` is not nil, so the following happens:
```
setInstallStatus(ext, nil)
setInstalledStatusConditionFailed(ext, err.Error())
```

A failed upgrade means that `installedBundle` is not nil, and `err` is not nil, so the following happens:
```
setInstallStatus(ext, installStatus)
setInstalledStatusConditionSuccess(ext, fmt.Sprintf("Installed bundle %s successfully", installedBundle.Image))
```

Reply:
If `err` is nil in here, then the result will be a message of "No install bundle found", rather than the installation error from `err`.